codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -69,47 +67,48 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
 
     override fun toMemberName(shape: MemberShape): String = escaper.escapeMemberName(shape.defaultName())
 
-    override fun byteShape(shape: ByteShape): Symbol = numberShape(shape, "Byte", "0.toByte()")
+    override fun byteShape(shape: ByteShape): Symbol = numberShape(shape, "Byte")
 
-    override fun integerShape(shape: IntegerShape): Symbol = numberShape(shape, "Int", "0")
+    override fun integerShape(shape: IntegerShape): Symbol = numberShape(shape, "Int")
 
     override fun intEnumShape(shape: IntEnumShape): Symbol = createEnumSymbol(shape)
 
-    override fun shortShape(shape: ShortShape): Symbol = numberShape(shape, "Short", "0.toShort()")
+    override fun shortShape(shape: ShortShape): Symbol = numberShape(shape, "Short")
 
-    override fun longShape(shape: LongShape): Symbol = numberShape(shape, "Long", "0L")
+    override fun longShape(shape: LongShape): Symbol = numberShape(shape, "Long")
 
-    override fun floatShape(shape: FloatShape): Symbol = numberShape(shape, "Float", "0f")
+    override fun floatShape(shape: FloatShape): Symbol = numberShape(shape, "Float")
 
-    override fun doubleShape(shape: DoubleShape): Symbol = numberShape(shape, "Double", "0.0")
+    override fun doubleShape(shape: DoubleShape): Symbol = numberShape(shape, "Double")
 
-    private fun numberShape(shape: Shape, typeName: String, defaultValue: String): Symbol =
-        createSymbolBuilder(shape, typeName, namespace = "kotlin").defaultValue(defaultValue).build()
+    private fun numberShape(shape: Shape, typeName: String): Symbol =
+        createSymbolBuilder(shape, typeName, namespace = "kotlin").build()
 
-    override fun bigIntegerShape(shape: BigIntegerShape?): Symbol = RuntimeTypes.Core.Content.BigInteger
+    // strip nullability from these runtime symbols as nullability is context dependent
+    override fun bigIntegerShape(shape: BigIntegerShape): Symbol = RuntimeTypes.Core.Content.BigInteger.asNonNullable()
 
-    override fun bigDecimalShape(shape: BigDecimalShape?): Symbol = RuntimeTypes.Core.Content.BigDecimal
+    override fun bigDecimalShape(shape: BigDecimalShape): Symbol = RuntimeTypes.Core.Content.BigDecimal.asNonNullable()
```

Comment:
I wasn't sure if it would or wouldn't break anything and saw no reason to make extensive changes since most of the symbols are used in a way that we wouldn't care about nullability anyway since they aren't used by `SymbolProvider` directly.